Zend/zend_language_scanner.l

```diff
@@ -1099,6 +1109,36 @@ NEWLINE ("\r"|"\n"|"\r\n")
 /* compute yyleng before each rule */
 <!*> := yyleng = YYCURSOR - SCNG(yy_text);
 
+<ST_IN_SCRIPTING,ST_LOOKING_FOR_SR_NAME,ST_LOOKING_FOR_VISIBILITY>{WHITESPACE}+ {
+	HANDLE_NEWLINES(yytext, yyleng);
+	return T_WHITESPACE;
+}
+
+<ST_LOOKING_FOR_SR_NAME>"->" {
+	return T_OBJECT_OPERATOR;
+}
+
+<ST_IN_SCRIPTING,ST_LOOKING_FOR_SR_NAME>"namespace"/"\\"|{OPTIONAL_WHITESPACE}"::" {
+	return T_NAMESPACE;
+}
+
+<ST_LOOKING_FOR_SR_NAME>{LABEL} {
+	yy_pop_state();
+	zend_copy_value(zendlval, yytext, yyleng);
+	return T_STRING;
+}
+
+<ST_LOOKING_FOR_SR_NAME>{ANY_CHAR} {
+	yyless(0);
+	yy_pop_state();
+	goto restart;
+}
+
+<ST_IN_SCRIPTING>{LABEL}/"\\"|{OPTIONAL_WHITESPACE}"::" {
```

Comment:
Nope, the rule wasn't dead. It was there for a **very** specific reason. But I just noticed it was causing a syntax regression. The following code is "valid" on  PHP 5.6:

```php
interface Thingable{}
class Thing implements\Thingable {}
// notice there is no space between implements and \
```
PHP 5.6 doesn't complain about it but the patched lexer was requiring white space between `implements` and `\`. I would consider it an accidental bug fix, but since it's a BC break the rule was removed.

PS: Some rules were simplified without regression by the last commit.